pkg/volume/aws_ebs/aws_util.go

```diff
@@ -107,3 +108,42 @@ func (util *AWSDiskUtil) DetachDisk(c *awsElasticBlockStoreCleaner) error {
 	}
 	return nil
 }
+
+func (util *AWSDiskUtil) DeleteVolume(d *awsElasticBlockStoreDeleter) error {
+	cloud := d.plugin.host.GetCloudProvider()
```

Comment:
nit: this code to fetch and cast the cloud provider gets repeated a lot, factor it out into a util method, if possible.